### PR TITLE
Ratings summary functionality

### PR DESCRIPTION
**Pull request by hollinwakefield**

### CODE REVIEW

File: client/src/components/ratings-reviews/RatingSummary.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+/* eslint-disable react/prefer-stateless-function */
+import React from 'react';
+
+class RatingSummary extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    if (!this.props.summary['5']) {
```
Comment (claire123c) [inline client/src/components/ratings-reviews/RatingSummary.jsx:15]: It looks like you're trying to copy the ratings data over to summary? Could you just do `let summary = this.props.ratings` ?

File: client/src/components/ratings-reviews/RatingSummary.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    let fiveStars = parseInt(this.props.summary['5']);
+    let fourStars = parseInt(this.props.summary['4']);
+    let threeStars = parseInt(this.props.summary['3']);
+    let twoStars = parseInt(this.props.summary['2']);
+    let oneStars = parseInt(this.props.summary['1']);
+    let totalRatings = fiveStars + fourStars + threeStars + twoStars + oneStars;
+    let averageRating = (((fiveStars * 5) + (fourStars * 4) + (threeStars * 3) + (twoStars * 2) + (oneStars)) / totalRatings).toFixed(1);
+
```
Comment (claire123c) [inline client/src/components/ratings-reviews/RatingSummary.jsx:38]: This code definitely runs correctly, but if you want to make your code more DRY, you could use a for loop to iterate through the elements. Something like this (psuedocoding): 
`let sum = 0
for (var key in summary) {
     sum += parseInt(summary[key])
}
let average = sum /  (Object.keys(summary).length)`

File: client/src/components/ratings-reviews/RatingSummary.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    return (
+      <div className="rating-summary">
+        <h3>Rating Summary</h3>
+        <p>5 Stars: {this.props.summary['5']}</p>
```
Comment (claire123c) [inline client/src/components/ratings-reviews/RatingSummary.jsx:42]: Optionally, if you want to make this code DRYer too, you could do something like (psuedocode):
`{Object.keys(summary).map((key) => (
   (use template literals)
     <p> '${key} Stars: ${summary[key]}' </p>
)`

It you want to order to be reversed like 5 to 1, you can use something like Array.reverse

File: client/src/components/ratings-reviews/Reviews.jsx [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
+import Ratings from './Ratings.jsx';
 import ReviewList from './ReviewList.jsx';
 
 // const getReviews = (review) => (<ReviewElement review={review}/>);
 
 class Reviews extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {'data': [{'body': 1}]};
+    this.state = {
```
Comment (claire123c) [inline client/src/components/ratings-reviews/Reviews.jsx:12]: I was researching what we talked about before. This solution works for now, but in the future if you have some kind of data coming in and it's not in the structure you want, you will encounter undefined. So in the future, I think to handle this issue, defaultProps or ternary/if statements (if something is defined, don't render) will be a better solution longterm.

**Review by claire123c — APPROVED** [review]
Looks good Hollin! Check out the comments made 😄

File: client/src/components/ratings-reviews/RatingSummary.jsx (hunk 1 above)
Comment (hollinwakefield) [reply client/src/components/ratings-reviews/RatingSummary.jsx:15]: Thanks for pointing this out

File: client/src/components/ratings-reviews/RatingSummary.jsx (hunk 2 above)
Comment (hollinwakefield) [reply client/src/components/ratings-reviews/RatingSummary.jsx:38]: Implemented, thank you

File: client/src/components/ratings-reviews/Reviews.jsx (hunk 4 above)
Comment (hollinwakefield) [reply client/src/components/ratings-reviews/Reviews.jsx:12]: This is good advice. Will refactor with this in mind if there's time.

**Pull request merged**